lib/Server/ReadStream.js

```diff
@@ -29,10 +29,11 @@ ReadStream.prototype._transform = function (chunk, encoding, next) {
 		this.pending = Buffer.concat([ this.pending, chunk ]);
 	}
 
-	while (this.pending.length >= 6) {
+	while (this.pending.length >= 12) {
 		var pkg_len = this.pending.readUInt16BE(4) + 6;
 
 		if (this.pending.length < pkg_len) return next();
+		if (pkg_len < 12) return next(); // every function code needs at least 4 additional bytes for "from" and "to". TODO: for WRITE_MULTIPLE_*, we could even check in more detail
```

Comment:
You can create custom functions with this library (since modbus protocol is scarce and leaves a lot of space and function codes to do it). This could be here but should be shorter.